### ISSUE TITLE
Connection Errors

### ISSUE THREAD

**Issue by kendog819** [SickRage/sickrage-issues#444]
These just started today..

![image](https://cloud.githubusercontent.com/assets/11353628/12188521/1a237be4-b584-11e5-8bfd-fddd0c3cee79.png)

![image](https://cloud.githubusercontent.com/assets/11353628/12188508/f575d7d8-b583-11e5-9360-146716dcb79e.png)

```
SEARCHQUEUE-DAILY-SEARCH :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [4eccee2]
```
Any help is appreciated

**Comment by SickRage2** [comment]
Thanks for the issue report! Before a real human comes by, please make sure your report has all the below criteria checked

- [ ] Include basic information: Branch/Commit, OS, What you did, What happened, What you expected
- [ ] Enable debug logging (be sure to disable after the bug is fixed)
- [ ] Post debug logs, either inline (for smaller logs) or using gist)

Please make sure you also read [how to create an issue](https://github.com/SickRage/sickrage-issues/blob/master/README.md) and followed all of the steps. 

The title should describe your issue. Having "SR not working" or "I get this bug" for 100 issues, isn't really helpful. We will close issues if there isn't enough information.

Sometimes the devs may seem like grunts and respond with short answers. This isn't (always) because the dev hates you, but because he's on mobile or busy fixing bugs. If something isn't clear, please let us know, and this bot may get updated to automatically answer you.

Thanks!

**Comment by labrys** [comment]
@kendog819 Sounds like reverse proxy issues.

**Comment by kendog819** [comment]
@labrys 

What does that mean, and how do I correct it?

Thanks

**Comment by labrys** [comment]
@kendog819 do you use a vpn, proxy, reverse proxy, firewall, or pfsense?  If so you have a configuration problem in one of those.

**Comment by Grasfer** [comment]
Also getting those errors.

**Comment by ericgoedtel** [comment]
@kendog819: It means that something is preventing you from accessing things over port 80. It could be a lot of things:
1. An intermediary device or application is blocking outbound port 80 from your host or network
2. An intermediary device or application is blocking access to those domains from your host or network

What platform is SickRage running on? Windows? Linux? NAS? There are different debugging options that we can look to try but it will depend upon what platform the app is being run from.

Long story short: It is not likely a problem with SickRage based on the information provided--but lets see if we can figure out what is going on.

**Comment by Grasfer** [comment]
Running sickrage in a jail under Freenas/Freebsd, usualy get those errors same time every day.


![Image of Config](http://i.imgur.com/baTLftJ.png)


```
2016-01-09 04:40:49 FINDPROPERS :: [NZBGeek] :: Connection error to getURL https://api.nzbgeek.info/api?apikey=04c1218954b7a95760ef6038a2cff6be&tvdbid=80379&season=9&maxage=3&cat=5030%2C5040&q=The.Big.Bang.Theory+PROPER%7CREPACK&limit=100&t=tvsearch&offset=0&ep=12 Error: u'error (\'Connection aborted.\', BadStatusLine("\'\'",))' [7fd6c98]
2016-01-09 03:14:37 SHOWQUEUE-REFRESH :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-09 03:12:31 SHOWQUEUE-REFRESH :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-09 03:11:28 SHOWQUEUE-REFRESH :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-09 03:10:51 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-09 03:10:17 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-09 03:09:42 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-09 03:09:08 SHOWQUEUE-FORCE-UPDATE :: Connection timed out accessing getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Read timed out. (read timeout=30)" [7fd6c98]
2016-01-09 03:07:50 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-09 03:06:17 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-09 03:05:36 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-09 03:04:56 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-09 03:04:10 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-09 03:02:31 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-09 03:01:20 DAILYSEARCHER :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
```

**Comment by rmbleeker** [comment]
I'm experiencing this as well. I'm not using any proxy's, VPN's and SickRage is running on a bare metal server, no VPS or jail or anything like that.

Branch: **master**
Commit: **7fd6c98aafe1f543bf8f96b2e370666be5412288**
Version: **v5.1**
OS: **Ubuntu 14.04 LTS** with kernel **4.2.0-22-generic**
Python version: **2.7.6**

```
2016-01-09 04:02:18 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
```

**Comment by OmgImAlexis** [comment]
@labrys I've been getting the same issue with news.md and xem for about 2 weeks now, figured it was just my internet randomly dropping.

**Comment by duramato** [comment]
Guys try and deleting on the cache folder , the sessions folder (if there is one) and the indexers

**Comment by ericgoedtel** [comment]
I have also observed a log error about thexem--I suspect this is a problem on thexem's end and not anything to do with SickRage. Casual HTTP status checks will periodically fail even outside of SickRage when trying to hit thexem's endpoint.

**Comment by kendog819** [comment]
@ericgoedtel 

I'm running QSickrage on a QNAP Nas..

![image](https://cloud.githubusercontent.com/assets/11353628/12217315/c8230826-b6ca-11e5-8b65-25e4d0314492.png)


Thanks =)

**Comment by OmgImAlexis** [comment]
@duramato I'm even getting this on my new install using Docker. Doesn't seem to be a caching issue.

**Comment by duramato** [comment]
Depends it may just be a timeout on the servers end as @ericgoedtel pointed out. If its that ( and most likely is ) theres nothing that can be one afaik..

**Comment by OmgImAlexis** [comment]
This is happening more and more now.
````
2016-01-10 05:45:57 CHECKVERSION :: Connection error to getURL http://sickrage.github.io/sickrage-news/news.md Error: u"error ('Connection aborted.', error(104, 'Connection reset by peer'))" [7fd6c98]
2016-01-10 03:29:19 SHOWQUEUE-REFRESH :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u'error (\'Connection aborted.\', BadStatusLine("\'\'",))' [7fd6c98]
````

**Comment by tutti21** [comment]
yeah this is happening every day

```js
2016-01-11 03:14:44 SHOWQUEUE-REFRESH :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-11 03:14:10 SHOWQUEUE-REFRESH :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-11 03:13:00 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-11 03:10:15 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-11 03:09:27 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-11 03:07:53 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-11 03:05:52 SHOWQUEUE-FORCE-UPDATE :: Connection timed out accessing getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Read timed out. (read timeout=30)" [7fd6c98]
2016-01-11 03:04:57 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-11 03:03:58 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-11 03:02:33 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
2016-01-11 03:01:53 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [7fd6c98]
```

**Comment by rmbleeker** [comment]
Same here, but I have to agree with @ericgoedtel that it's probably on the end of TheXem, so nothing the SickRage team can do about it except maybe suppress the warnings in the log. I'm not sure what TheXem does for SickRage but everything seems to be working fine, so I'm guessing a connection is made at least some of the time.

**Comment by duramato** [comment]
Xem gives the shows the scene numbering

**Comment by ericgoedtel** [comment]
I think that we may have two different things going on. The original reporter is referring to an outage for a specific set of trackers but we seem to have focused on the issue with TheXem (my fault, oops).

@kendog819: Back to your issue: You are seeing retry rejections--basically that some system is saying that you are trying to access it too rapidly.

**TL;DR**: You are likely seeing HTTP rate limiting across multiple domains--we need to see what is between you and the HTTP servers. In this case, you specified that you are using an HTTP proxy. I suspect that the HTTP proxy is dumping you because you are attempting to send too much traffic through.

Longer version:

If you were seeing this on one host (such as a tracker or indexer) then I would say "Stop hitting them so frequently." You seem to be hitting a limiter when accessing multiple domains simultaneously. Barring something extraordinary--its unlikely that your local network is accessing two or more domains with such rapidity as to receive an HTTP 429 (or whatever applicable HTTP code is causing the issue).

**Comment by kendog819** [comment]
@ericgoedtel 

Thanks for the info... What do you suggest I do? I currently have SR set to:

![image](https://cloud.githubusercontent.com/assets/11353628/12237485/ec5942d6-b84c-11e5-92ff-2c16cbb78ac2.png)

Is this too short of a time?

**Comment by duramato** [comment]
@kendog819 YES , thats too little try a bigger time..

**Comment by ericgoedtel** [comment]
@kendog819 More specifically--you need to check what rate limits are in place on your proxy. The proxy is limiting you. I cannot seem to find any documentation on limits from your specified proxy but I am sure that there is one.

**Comment by OmgImAlexis** [comment]
@fernandog I'm still getting errors with the NEWS.md page and XEM and I'm not using any proxy at all. This has been going on for the last few weeks.

**Comment by duramato** [comment]
It can well be a problem with your internet connection/isp

**Comment by OmgImAlexis** [comment]
@duramato then why haven't I had this issue for the last 2+ years that I've been using Sickrage? @miigotu can you look into this please.

**Comment by duramato** [comment]
Thats like saying i have had power for the last 2 years why is it failing now? But can well explain why only some people have that ( besides the usuall time outs)

**Comment by ericgoedtel** [comment]
Wouldn't it be more appropriate to open a new req or something for the XEM-specific issue? Original request was opened because of rate limiting. XEM issue seems to be caused by timeouts to the host. Ya'll are the collab so that's your call but the original issue is solved :)

**Comment by OmgImAlexis** [comment]
@duramato we've been changing a bunch of code since we split from sickragetv I'm assuming it all because of something we've changed and not from internet dropouts, etc. I'll just on IRC tomorrow and see if I can get it sorted.

**Comment by kendog819** [comment]
@ericgoedtel 

![image](https://cloud.githubusercontent.com/assets/11353628/12276493/1a5f1762-b944-11e5-83bf-5fde3cf4036b.png)

Any other suggestions?

Thanks

**Comment by miigotu** [comment]
Sure, a slow vpn?

**Comment by kendog819** [comment]
I typically get ~2 mbps actual d/l speed thru the proxy. So I don't think that is it.

**Comment by miigotu** [comment]
Its not about the overall speed if it its the initial response time. Sure a connection can even be 1000MB/s but the initial connection can take a long time to set up.

**Comment by kendog819** [comment]
Is there something I can use to test the proxies so that I am using the best one? What am I looking for? lowest ping time?

**Comment by ericgoedtel** [comment]
You might want to consider taking a packet capture and see who is to blame. Is SickRage taking too long to transmit a packet or is the tracker taking too long to respond? Still looking to see a full sickrage.log file so that I can see the specific error. The others on this thread are for TheXEM and not your tracker.

Another curiosity would be to determine if the HTTP URL printed in the log is accessible off of the VPN. Can you disable the VPN for a time to at least rule it out?

**Comment by ward0** [comment]
no proxy , vpn , direct connection trough router/modem , no firewall.

port 80 is allowed ofcourse.

```
2016-01-29 03:07:16 SHOWQUEUE-REFRESH :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [f568d15]
2016-01-29 03:06:31 SHOWQUEUE-REFRESH :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [f568d15]
2016-01-29 03:05:51 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [f568d15]
2016-01-29 03:05:15 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [f568d15]
2016-01-29 03:04:31 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [f568d15]
2016-01-29 03:03:46 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [f568d15]
2016-01-29 03:02:56 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [f568d15]
2016-01-29 03:00:58 SHOWQUEUE-FORCE-UPDATE :: Connection error to getURL http://thexem.de/map/havemap?origin=tvdb Error: u"error HTTPConnectionPool(host='thexem.de', port=80): Max retries exceeded with url: /map/havemap?origin=tvdb (Caused by ConnectTimeoutError(, 'Connection to thexem.de timed out. (connect timeout=30)'))" [f568d15]
```

**Comment by labrys** [comment]
@ward0 xem timeouts are to be expected as several people have mentioned.  And it does not appear that you have any other timeouts listed.

**Comment by ericgoedtel** [comment]
Can anyone take a packet capture for this? Would be nice to see. I suspect these timeouts are due to thexem and not SickRage.

**Comment by duramato** [comment]
@ericgoedtel they are for sure especiay if your show update is at 2am , xem is always under ddos from all the users

**Comment by ericgoedtel** [comment]
@duramato Hooray! Hopefully no one else will come back with issues with thexem to this thread.

@kendog819  We keep getting away from your question because of thexem issues. Have you had an opportunity to review my last response to you? pcap would be good if you're still running into issues connecting to your endpoints.
